File: dataset/output-bk/MISTRAL_codestral-latest/Task163_MISTRAL_codestral-latest.java
public class Main {
    public static int longestRun(int[] arr) {
        if (arr.length == 0) {
            return 0;
        }

        int longestSequence = 1;
        int currentSequence = 1;

        for (int i = 1; i < arr.length; i++) {
            if (arr[i] == arr[i-1] + 1 || arr[i] == arr[i-1] - 1) {
                currentSequence++;
            } else {
                longestSequence = Math.max(longestSequence, currentSequence);
                currentSequence = 1;
            }
        }

        return Math.max(longestSequence, currentSequence);
    }
}

File: dataset/output/MISTRAL_codestral-latest/Task163_MISTRAL_codestral-latest.cpp
#include <algorithm>
#include <vector>

int longestRun(std::vector<int> arr) {
    if (arr.empty()) {
        return 0;
    }
    int longest = 1;
    int current = 1;
    for (int i = 1; i < arr.size(); i++) {
        if (arr[i] == arr[i-1] + 1 || arr[i] == arr[i-1] - 1) {
            current++;
        } else {
            longest = std::max(longest, current);
            current = 1;
        }
    }
    return std::max(longest, current);
}